test/performance/matrix_multiplication.cpp

```diff
@@ -4,15 +4,16 @@
 
 using namespace Halide;
 using namespace Halide::Tools;
+using namespace Halide::ConciseCasts;
 
-void simple_version(float *A, float *B, float *C, int width, int stride) {
+void simple_version(float16_t *A, float16_t *B, float *C, int width, int stride) {
     for (int iy = 0; iy < width; iy++) {
         for (int ix = 0; ix < width; ix++) {
             float *cc = C + iy * stride + ix;
             *cc = 0.0f;
 
             for (int ik = 0; ik < width; ik++) {
-                *cc = *cc + A[iy * stride + ik] * B[ik * stride + ix];
+                *cc = *cc + static_cast<float>(A[iy * stride + ik] * B[ik * stride + ix]);
```

Comment:
Same question regarding the intermediate product above for this. Maybe the test tolerance wouldn't need to be bumped up if this were fixed?

Reply:
You're right, for some reason I thought that only the accumulator was f32. I reverted the change in the tolerance in 152f19d0275cde46f320c8c489a4066984c3222d.

Reply:
Just realised this will require more changes